pkg/webhook/handler.go

```diff
@@ -23,8 +23,10 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 )
 
+type Func func(ctx context.Context, req admission.Request) admission.Response
+
 type Handler interface {
-	OnCreate(ctx context.Context, req admission.Request, client client.Client, decoder *admission.Decoder) admission.Response
-	OnDelete(ctx context.Context, req admission.Request, client client.Client, decoder *admission.Decoder) admission.Response
-	OnUpdate(ctx context.Context, req admission.Request, client client.Client, decoder *admission.Decoder) admission.Response
+	OnCreate(client client.Client, decoder *admission.Decoder) Func
```

Comment:
I was shadowing the type to don't lock in to the underlying implementation since we're not using the Kubebuilder/Operator SDK native implementation, just to avoid breaking changes for the upcoming future in case of signature change.